stdlib/public/core/Runtime.swift.gyb

```diff
@@ -260,14 +253,14 @@ public func _swift_stdlib_atomicFetch${operation}Int(
   operand: Int) -> Int {
 #if arch(i386) || arch(arm)
   return Int(Int32(bitPattern:
-    _swift_stdlib_atomicFetch${operation}UInt32(
-      object: UnsafeMutablePointer(target),
-      operand: UInt32(bitPattern: Int32(operand)))))
+      _swift_stdlib_atomicFetch${operation}UInt32(
+        object: unsafeBitCast(target, to: UnsafeMutablePointer<UInt32>.self),
+        operand: UInt32(bitPattern: Int32(operand)))))
 #elseif arch(x86_64) || arch(arm64) || arch(powerpc64) || arch(powerpc64le) || arch(s390x)
   return Int(Int64(bitPattern:
-    _swift_stdlib_atomicFetch${operation}UInt64(
-      object: UnsafeMutablePointer(target),
-      operand: UInt64(bitPattern: Int64(operand)))))
+      _swift_stdlib_atomicFetch${operation}UInt64(
+        object: unsafeBitCast(target, to: UnsafeMutablePointer<UInt64>.self),
+        operand: UInt64(bitPattern: Int64(operand)))))
```

Comment:
FYI emacs is intentionally following a convention here. I'm not picky about the convention as long as swift mode can be fixed.